index.d.ts

```diff
@@ -10,6 +10,13 @@ import {
 
 declare namespace contextMenu {
 	interface Labels {
+		/**
+		The default label for the `Look Up [selection]` menu item is dynamically generated depending on the selected text when the menu is shown.
+		If a value is set here, the provided static value will be used instead.
```

Comment:
A better solution would be to handle this for the user, so if they specify a label of `foo`, the end result would be `foo “Some selection“`.

Reply:
Hmmm, I'm not sure that's a good solution – it depends what the user's reason is for overriding the label  in the first place.

If it's for i18n, this would  be problematic, because the order wouldn't be the same in every language (some languages, such as Korean or Nepali, would put the word being looked up first), and nor would the type of quote marks (eg, French uses `«…»`, Polish uses `„…”`).

If it's for any other reason, maybe the user actually *wants* a static string; they might not want anything appended to it.

If we want to keep it dynamic for localisations,  I think a better way would be for `label` to be able to optionally accept a callback instead of a string, which would be called with `props` as an argument. Then the callback could generate whatever string is wanted and return it.
What do you think of this idea?

Reply:
So an example usage might be:
```js
labels: {
	copy: 'Copiar',
	lookUpSelection: props => `Consultar “${cliTruncate(props.selectionText.trim(), 25)}”`
}
```

Reply:
We can just let it support a replacement. So the user could specify `foo {selection}` and have `{selection}` replaced with the dynamic string.

Function callback would be useful too, but that's a separate feature. https://github.com/sindresorhus/electron-context-menu/pull/75

Reply:
> We can just let it support a replacement. So the user could specify `foo {selection}` and have `{selection}` replaced with the dynamic string.

Yes, that's not a bad option.
Pity JavaScript doesn't have native support for template strings…
I'll play with some ideas in a bit.

> Function callback would be useful too, but that's a separate feature. #75

:+1:

Reply:
Done. I've made it so the `{selection}` placeholder can be used in any menu title, as it could easily be useful in other places. (Besides, the code's cleaner without any special-casing.)